src/main/java/com/meilisearch/sdk/json/GsonJsonHandler.java

```diff
@@ -16,25 +19,27 @@ public GsonJsonHandler(Gson gson) {
 
     @Override
     public String encode(Object o) throws Exception {
-        if (o.getClass() == String.class) {
+        if (o != null && o.getClass() == String.class) {
             return (String) o;
         }
+        if (o != null && o.getClass() == Key.class) {
```

Comment:
Can you put a `TODO: review later` here on this line?

We cannot create those levels of specialization inside of this class because this is an infinite job if other classes start to have similar issues to be handled.